engines/director/lingo/lingo-preprocessor.cpp

```diff
@@ -26,17 +26,36 @@
 
 namespace Director {
 
-Common::String preprocessWhen(Common::String in, bool *changed);
-
-bool isspec(char c) {
-	return strchr("-+*/%%^:,()><&[]=", c) != NULL;
+bool isspec(Common::u32char_type_t c) {
+	switch (c) {
+	case '-':
```

Comment:
I am not sure it will work reliably, all these characters are ASCII, and you're checking against u32char_type_t.

Reply:
Changed back to the previous implementation with a bounds check on the input.